design/pv_backup_info.md

```diff
@@ -0,0 +1,174 @@
+
+    SnapshotHandle      string       // The actual snapshot ID. It can be the cloud provider's snapshot ID or the file-system uploader's snapshot.
+    Size                int64        // The snapshot corresponding volume size. Some of the volume backup methods cannot retrieve the data by current design, for example, the Velero native snapshot.
+    StartTimestamp      *metav1.Time // Snapshot starts timestamp.
+
+    DataMoverInfo       DataMoverInfo
+    NativeSnapshotInfo  VeleroNativeSnapshotInfo
+    PVBInfo             PodVolumeBackupInfo
+}
+
+// DataMoverInfo is used for displaying the snapshot data mover and PodVolumeBackup snapshot status.
+type DataMoverInfo struct {
+
+    DataMover        string    // The data mover used by the backup. The valid values are `velero` and ``(equals to `velero`).
+    UploaderType     string    // The type of the uploader that uploads the snapshot data. The valid values are `kopia` and `restic`. It's useful for file-system backup and snapshot data mover.
+    RetainedSnapshot string    // The name or ID of the snapshot associated object(SAO).
+}
+
+// VeleroNativeSnapshotInfo is used for displaying the Velero native snapshot status.
+type VeleroNativeSnapshotInfo struct {
+    VolumeType string    // The cloud provider snapshot volume type.
+    VolumeAZ   string    // The cloud provider snapshot volume's availability zones.
+    IOPS       string    // The cloud provider snapshot volume's IOPS.
+}
+
+// PodVolumeBackupInfo is used for displaying the PodVolumeBackup snapshot status.
+type PodVolumeBackupInfo struct {
+    UploaderType  string    // The type of the uploader that uploads the data. The valid values are `kopia` and `restic`. It's useful for file-system backup and snapshot data mover.
+    VolumeName    string // The PVC's corresponding volume name used by Pod
+    PodName       string // The Pod name mounting this PVC. The format should be <namespace-name>/<pod-name>.
+}
+```
+
+To make Velero support multiple versions of VolumeInfo, Velero needs to create a structure to include the version and the versioned volume information together. The information is persisted in the metadata file during backup creation. When reading the VolumeInfo metadata file, Velero reads the version information first by un-marshalling the metadata file by structure `VolumeInfoVersion`, then decide to use which version of the VolumeInfos according to the version value.
+
+If there is need to introduce a non compatible change to the VolumeInfo, a new version of `VolumeInfos` and `VolumeInfo` are needed. For example, version 2.0 is created, then `VolumeInfosV20` and `VolumeInfoV20` structures are needed.
+
+``` golang
+type VolumeInfoVersion struct {
+    Version string 
+}
+
+type VolumeInfosV10 struct {
+    Infos   []VolumeInfoV10
+    Version string // VolumeInfo structure's version information.
+}
+```
+
+### How the VolumeInfo array is generated.
+
```

Comment:
Here you didn't really describe how this array is **generated** 
To specify how it is **generated**  you need to provide information like how to extend the capabilities of `skippedPVTracker`, etc.
One challenge I can think about is how to aggregate information in different categories into one list, esp considering we may process different PVs in parallel.

Reply:
The plan is to generate the array in the function `persistBackup`.
The function `persistBackup` has `backup *pkgbackup.Request` in parameters.
From it, the `VolumeSnapshots`, `PodVolumeBackups`, `CSISnapshots`, `itemOperationsList`, and `SkippedPVTracker` can be read. All of them will be iterated and merged into the `VolumeInfo` array.